decent-dao/fractal-interface
# Update parameters splitting logic and validation

## Description


This PR updated the way in which function signatures are processed on the Proposal Create page. And a refactor to dry up some of the code on the Transaction.txs page.

## Notes
NO QUOTES ARE NEEDED FOR STRINGS OR ADDRESSES in the parameter input. Use a comma to separate parameter values. We will need to deal with escaping a comma if we want to be able to pass that in a string.

## Issue / Notion doc (if applicable)
https://github.com/decent-dao/fractal-interface/issues/797


## Testing
1) Create a proposal with various inputs. Test `address`, `uint256`, and `string`
2) Enter the parameters with the uint256 having 18+ digits
3) Make sure no error validation pops up to any properly formatted signatures / parameter.
4) Create proposal, check that decoded values are correct
5) Test a proposal with a function with no parameters.


## Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/99622829/208186610-6c2bc992-dc66-46e0-8ccd-4baad5aa4688.png)
